lib/with_test_parser.rb

```diff
@@ -0,0 +1,8 @@
+require 'yaml'
+require 'active_support/core_ext/hash'
+
+module WithTestParser
+  def parse_test(request)
+    (YAML.load request[:test]).deep_symbolize_keys
```

Comment:
Instead of using `deep_symbolize_keys`, you could use `with_indifferent_access` which makes using strings or symbols interchangable when accessing values.